src/transfer-handler.js

```diff
@@ -11,23 +14,37 @@ class TransferHandler {
 
   onPlugin (prefix) {
     this.main.getPlugin(prefix).on('incoming_prepare', (transfer) => {
-      const destination = IlpPacket.deserializeIlpPayment(Buffer.from(transfer.ilp, 'base64'))
-      const onwardTransfer = this.main.quoter.findHop(destination.account, destination.amount)
-      const onwardPlugin = this.main.getPlugin(onwardTransfer.ledger)
-      this.outstanding[transfer.executionCondition] = transfer
-      onwardPlugin.sendTransfer(Object.assign(onwardTransfer, {
-        id: uuid(),
-        from: onwardPlugin.getAccount(), // see https://github.com/interledger/rfcs/issues/289
-        // to set by findHop
-        ledger: onwardPlugin.getInfo().prefix, // see https://github.com/interledger/rfcs/issues/289
-        // amount set by findHop
-        ilp: transfer.ilp,
-        noteToSelf: {},
-        executionCondition: transfer.executionCondition,
-        expiresAt: new Date(new Date(transfer.expiresAt).getTime() + MIN_MESSAGE_WINDOW).toISOString(),
-        custom: {}
-      })).catch(err => {
-        console.error('forwarding failed', err.message)
+      this.main.voucher.checkVouch(transfer.from, transfer.amount).then((answer) => {
+        if (answer || this.main.getPlugin(prefix).isPrivate) {
+          const packet = IlpPacket.deserializeIlpPacket(Buffer.from(transfer.ilp, 'base64'))
+          let onwardTransfer
+          if (packet.type === TYPE_EXACT) {
+             onwardTransfer = this.main.quoter.findHop(packet.data.account, packet.data.amount)
+          } else if (packet.type === TYPE_FORWARDED) {
+             onwardTransfer = this.main.quoter.findForwardedHop(packet.data.account, transfer.amount)
+          }
+          const onwardPlugin = this.main.getPlugin(onwardTransfer.ledger)
+          this.outstanding[transfer.executionCondition] = transfer
+          onwardPlugin.sendTransfer(Object.assign(onwardTransfer, {
+            id: uuid(),
+            from: onwardPlugin.getAccount(), // see https://github.com/interledger/rfcs/issues/289
+            // to set by findHop
+            ledger: onwardPlugin.getInfo().prefix, // see https://github.com/interledger/rfcs/issues/289
+            // amount set by findHop
+            ilp: transfer.ilp,
+            noteToSelf: {},
+            executionCondition: transfer.executionCondition,
+            expiresAt: new Date(new Date(transfer.expiresAt).getTime() + MIN_MESSAGE_WINDOW).toISOString(),
+            custom: {}
+          })).catch(err => {
+            console.error('forwarding failed', err.message)
+          })
+        } else {
+          this.main.getPlugin(prefix).rejectIncomingTransfer(transfer, {
+            code: 'L53',
```

Comment:
Is there an appropriate error code from https://github.com/interledger/rfcs/blob/master/0003-interledger-protocol/0003-interledger-protocol.md#ilp-error-codes that could be used?

Reply:
Not really I think, which one were you thinking of?

Reply:
I didn't have one in mind, but the `"L"` prefix isn't used anywhere afaik. It should either be `"F"`, `"T"`, or `"R"` depending on what type of error it is. And probably `99` to indicate it is an application error.

Reply:
This code no longer exists.